pkg/reconciler/ingress/ingress_test.go

```diff
@@ -247,61 +252,85 @@ func withAnnotations(ans map[string]string) ingressCreationOption {
 	}
 }
 
-func withPreferHeaderPaths(isAlwaysAsync bool) ingressCreationOption {
-	return func(ing *v1alpha1.Ingress) {
-		// Get ingress name depending on async property
-		var ingName string
-		var ingNamespace string
-		if isAlwaysAsync {
-			ingName = ingAlwaysAsync.Name
-			ingNamespace = ingAlwaysAsync.Namespace
-		} else {
-			ingName = ingWithAsyncAnnotation.Name
-			ingNamespace = ingWithAsyncAnnotation.Namespace
-		}
-		splits := make([]v1alpha1.IngressBackendSplit, 0, 1)
-		splits = append(splits, v1alpha1.IngressBackendSplit{
-			IngressBackend: v1alpha1.IngressBackend{
-				ServiceName:      ingName + asyncSuffix,
-				ServiceNamespace: ingNamespace,
-				ServicePort:      intstr.FromInt(80),
+func expectedCreatedIng(namespace, name string, status v1alpha1.IngressStatus, isAlwaysAsync bool) *v1alpha1.Ingress {
+	thePaths := []netv1alpha1.HTTPIngressPath{}
+	if isAlwaysAsync {
+		thePaths = append([]netv1alpha1.HTTPIngressPath{{
```

Comment:
why not just define the array here directly (I dont think the append is doing anything helpful?)